contrib/cmd/runc-dmz/runc-dmz.c

```diff
@@ -0,0 +1,11 @@
+#include <unistd.h>
+
+extern char **environ;
+
+int main(int argv, char **args)
+{
+    if (argv > 0) {
+        return execve(args[0], args, environ);
+    }
+    return 0;
+}
```

Comment:
I guess we can optimize the binary footprint by eliminating glibc and executing “syscall” instruction directly, but that may happen in follow-up PRs

Reply:
If we build this binary with musl, you get a 13KB binary. With glibc it's 1.1MB. We will need to adjust our build system for this -- but there is a separate issue that distributions might find building a single binary with musl quite difficult. Speaking from the SUSE side, I don't think we even package musl for SLES so we would need to build with glibc (meaning the binary would be 1.1MB in size -- almost 100 times larger).

As you say, this can be done in a future PR.

Reply:
Rewriting this in asm will be much easier than complicating libc dependency

Reply:
`syscall` doesn't reduce the binary size unfortunately (also, libcs provide the syscall numbers for all syscalls in `<sys/syscall.h>`).

Reply:
https://gist.github.com/Zheaoli/f37bc1fb04917fdfac36d644ee69f7e9
I made a demo for amd64. The binary file size is 4.9k.

Reply:
I think maybe there still have some design problems for `runc-dmz` proposal.
Please see: https://github.com/opencontainers/runc/pull/3987#issuecomment-1698878930
But I can't confirm, if you can check there is no problem for `runc-dmz` propsal, it will be appreciated. Because I don't want to waste contributor's time for a wrong proposal.